Modules/Core/Common/include/itkDICOMOrientation.h

```diff
@@ -0,0 +1,280 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkDICOMOrientation_h
+#define itkDICOMOrientation_h
+
+#include "ITKCommonExport.h"
+#include "itkImageBase.h"
+#ifndef ITK_FUTURE_LEGACY_REMOVE
+#include "itkSpatialOrientation.h"
+#endif
+#include <map>
+#include <string>
+
+namespace itk
+{
+
+/** \class DICOMOrientation
+ * \brief A supporting class for DICOMOrientImageFilter.
+ *
+ *  Defines enums to for patient orientation in a compatible way with DICOM.
+ *
+ *  Instances hold the patient orientation enum and allow conversion to and from enums, string and direction cosine
+ * matrices. Conversions from a direction cosine matrix is approximated with the orientation of the closes axes.
+ *
+ * \ingroup ITKCommon
+ */
+class ITKCommon_EXPORT DICOMOrientation
+{
+public:
+  constexpr static unsigned int Dimension = 3;
+  using DirectionType = typename ImageBase<Dimension>::DirectionType;
+  constexpr static unsigned int ImageDimension = Dimension;
+
+#ifndef ITK_FUTURE_LEGACY_REMOVE
+  using LegacyOrientationType = SpatialOrientationEnums::ValidCoordinateOrientations;
+#endif
+
+  enum class CoordinateEnum : uint8_t
+  {
+    UNKNOWN = 0,
+    Left = 2,  ///< 0b0010
+    RightToLeft = 2,
+    Right = 3,
+    LeftToRight = 3,
+    Anterior = 4,  ///< front - 0b0100
+    PosteriorToAnterior = 4,
+    Posterior = 5, ///< back
+    AnteriorToPosterior = 5,
+    Superior = 8,   ///< above - 0b1000
+    InferiorToSuperior = 8,
+    Inferior = 9,  ///< bottom
+    SuperiorToInferior = 9,
+  };
+
+private:
+  enum class CoordinateMajornessTermsEnum : uint8_t
+  {
+    PrimaryMinor = 0,
+    SecondaryMinor = 8,
+    TertiaryMinor = 16
+  };
+
+  constexpr static uint32_t
+  toOrientation(const CoordinateEnum primary, const CoordinateEnum secondary, const CoordinateEnum tertiary)
```

Comment:
It may be just a matter of style, but it appears that "static constexpr" would be preferable here, instead of  "constexpr static".  See  "static constexpr unsigned long is C++’s “lovely little old French whittling knife”, by Arthur O’Dwyer: https://quuxplusone.github.io/blog/2021/04/03/static-constexpr-whittling-knife/

Reply:
-  Related to pull request #3144 by Lee Newberg (@Leengit)

Reply:
done